pkg/broker/binding_operations.go

```diff
@@ -139,20 +141,36 @@ func userFromParams(bindingID string, password string, rawParams []byte) (*atlas
 		}
 	}
 
-	// Set binding ID as username and add password.
-	params.User.Username = bindingID
-	params.User.Password = password
-
-	// If no role is specified we default to read/write on any database.
-	// This is the default role when creating a user through the Atlas UI.
-	if len(params.User.Roles) == 0 {
-		params.User.Roles = []atlas.Role{
-			atlas.Role{
-				Name:         "readWriteAnyDatabase",
-				DatabaseName: "admin",
-			},
+	// If the plan ID is specified we construct the provider object from the service and plan.
+	// The plan ID is required for binding
+	if planID != "" {
```

Comment:
This is not applicable for bindings.

Reply:
Yep, I see it now. It's always required according to the spec.